src/components/Button.res

```diff
@@ -698,107 +699,109 @@ let make = (
 
   let newThemeGap = ""
 
-  <AddDataAttributes attributes=[(dataAttrKey, dataAttrStr)]>
-    <button
-      type_
-      disabled=dis
-      ref={parentRef->ReactDOM.Ref.domRef}
-      onKeyUp={e => e->ReactEvent.Keyboard.preventDefault}
-      onKeyPress={e => {
-        if !onEnterPress {
-          e->ReactEvent.Keyboard.preventDefault
-        }
-      }}
-      className={`flex group ${customButtonStyle->String.includes("justify")
-          ? ""
-          : "justify-center"} ${relativeClass} ${heightClass} ${newThemeGap} ${conditionalButtonStyles} items-center ${borderStyle}  ${textColor} ${cursorType} ${paddingClass} ${lengthStyle} ${customButtonStyle}  ${customTextOverFlowClass}`}
-      onClick=handleClick>
-      {if buttonState == Loading {
-        <span className={iconPadding}>
-          <span className={`flex items-center mx-2 animate-spin`}>
-            <Loadericon size=iconSize iconColor=?loaderIconColor />
+  <UIUtils.RenderIf condition={access !== AuthTypes.NoAccess}>
+    <AddDataAttributes attributes=[(dataAttrKey, dataAttrStr)]>
+      <button
+        type_
+        disabled=dis
+        ref={parentRef->ReactDOM.Ref.domRef}
+        onKeyUp={e => e->ReactEvent.Keyboard.preventDefault}
+        onKeyPress={e => {
+          if !onEnterPress {
+            e->ReactEvent.Keyboard.preventDefault
+          }
+        }}
+        className={`flex group ${customButtonStyle->String.includes("justify")
+            ? ""
+            : "justify-center"} ${relativeClass} ${heightClass} ${newThemeGap} ${conditionalButtonStyles} items-center ${borderStyle}  ${textColor} ${cursorType} ${paddingClass} ${lengthStyle} ${customButtonStyle}  ${customTextOverFlowClass}`}
+        onClick=handleClick>
+        {if buttonState == Loading {
+          <span className={iconPadding}>
+            <span className={`flex items-center mx-2 animate-spin`}>
+              <Loadericon size=iconSize iconColor=?loaderIconColor />
+            </span>
           </span>
-        </span>
-      } else {
-        switch leftIcon {
+        } else {
+          switch leftIcon {
+          | FontAwesome(iconName) =>
+            <span className={`flex items-center ${iconColor} ${iconMargin} ${iconPadding}`}>
+              <Icon
+                className={`align-middle ${strokeColor} ${iconBorderColor}`}
+                size=iconSize
+                name=iconName
+              />
+            </span>
+          | Euler(iconName) =>
+            <span className={`flex items-center ${iconColor} ${iconMargin} ${eulerIconPadding}`}>
+              <Icon className={`align-middle ${strokeColor}`} size=iconSize name=iconName />
+            </span>
+          | CustomIcon(element) =>
+            <span className={`flex items-center ${iconMargin}`}> {element} </span>
+          | _ => React.null
+          }
+        }}
+        {switch text {
+        | Some(textStr) =>
+          if textStr !== "" {
```

Comment:
use util function so check empty string

Reply:
Moved this function to LogicUtils and used that